sql/2021/mobile-web/ecommerce_with_type_search.sql

```diff
@@ -0,0 +1,50 @@
+#standardSQL
+# Ecommerce sites using type=search inputs
+CREATE TEMPORARY FUNCTION getSearchInputStats(payload STRING)
+RETURNS STRUCT<has_inputs BOOLEAN, has_search_inputs BOOLEAN> LANGUAGE js AS '''
+  try {
+    const almanac = JSON.parse(payload);
+    const search_node_index = almanac.input_elements.nodes.findIndex((node) => {
+      return node.type.toLowerCase() === 'search';
+    });
+
+    return {
+      has_inputs: almanac.input_elements.total > 0,
+      has_search_inputs: search_node_index >= 0,
```

Comment:
Why `>=`?

Reply:
`findIndex` will return the array index it finds a match, otherwise -1